backend/compact-connect/lambdas/python/common/cc_common/data_model/provider_record_util.py

```diff
@@ -224,6 +233,137 @@ def assemble_provider_records_into_api_response_object(provider_records: list[di
+            datetime_of_expiration_trigger = datetime.combine(
+                day_after_expiration, datetime.min.time(), tzinfo=config.expiration_resolution_timezone
+            )
+            effective_date_time = datetime.combine(
+                update['effectiveDate'], datetime.min.time(), tzinfo=config.expiration_resolution_timezone
+            )
+            if datetime_of_expiration_trigger <= effective_date_time:
+                enriched_history.append(
+                    {
+                        'type': 'privilegeUpdate',
+                        'updateType': UpdateCategory.EXPIRATION,
+                        'providerId': privilege['providerId'],
+                        'compact': privilege['compact'],
+                        'jurisdiction': privilege['jurisdiction'],
+                        'licenseType': privilege['licenseType'],
+                        'effectiveDate': date_of_expiration,
+                        'createDate': datetime_of_expiration_trigger.astimezone(UTC),
+                        'previous': {},
+                        'updatedValues': {},
+                        'dateOfUpdate': datetime_of_expiration_trigger.astimezone(UTC),
+                    }
+                )
+        # Inject expiration event if currently expired
+        privilege_date_of_expiration = privilege['dateOfExpiration']
+
+        privilege_day_after_expiration = privilege_date_of_expiration + timedelta(days=1)
+        privilege_datetime_of_expiration_trigger = datetime.combine(
+            privilege_day_after_expiration, datetime.min.time(), tzinfo=config.expiration_resolution_timezone
+        )
+
+        if privilege_datetime_of_expiration_trigger <= now.astimezone(config.expiration_resolution_timezone):
+            enriched_history.append(
+                {
+                    'type': 'privilegeUpdate',
+                    'updateType': UpdateCategory.EXPIRATION,
+                    'providerId': privilege['providerId'],
+                    'compact': privilege['compact'],
+                    'jurisdiction': privilege['jurisdiction'],
+                    'licenseType': privilege['licenseType'],
+                    'effectiveDate': privilege_date_of_expiration,
+                    'createDate': privilege_datetime_of_expiration_trigger.astimezone(UTC),
+                    'previous': {},
+                    'updatedValues': {},
+                    'dateOfUpdate': privilege_datetime_of_expiration_trigger.astimezone(UTC),
+                }
+            )
+
+        return sorted(enriched_history, key=lambda x: x['effectiveDate'])
+
+    @staticmethod
+    def construct_simplified_privilege_history_object(privilege_data: list[dict]) -> dict:
+        """
+        Construct a simplified list of history events to be easily consumed by the front end
+        :param privilege_data: All of the records relative to the privilege: the privilege, updates, and adverse actions
```

Comment:
maybe? is `privilege_data` the raw data or the augmented synthetic data?

```suggestion
        :param privilege_data: All of the records associated with the privilege: the privilege, updates, and adverse actions
```